[ BE ] (docs) 9WEEK

02yebin opened the pull request:
[ BE ] (docs) 9WEEK
9주차 스피링 MVC 5-7장 알아둬야하는 부분만 요약하였습니다.

저번주에 PR보냈는데 왜 안됐는지 모르겠네요,,

02yebin committed:
[BE] (REFACTOR) lastweek commit 비교용

지난주에 작성한 코드를 수정, 보완할 것이기에 비교용으로 올려둡니다.

02yebin committed:
[BE] (REFACTOR) lastweek commit 비교용

지난주에 작성한 코드를 수정, 보완할 것이기에 비교용으로 올려둡니다.

02yebin committed:
[BE] (DOCS) theory summary

알아야하는 이론을 조사해 요약했습니다.

02yebin committed:
[BE] (REFACTOR) DiscountPolicy class

할인 정책을 바꾸어주었는데 아직 고정 금액 할인과 정률 할인을 구분하는 방법을 찾지못함

02yebin committed:
뭔가 잘못됐다..

무언가 잘못되었음을 느꼈지만 여기서 더 건들이면 돌이킬수없을것 같아 그대로 커밋합니다..

02yebin committed:
Merge branch 'yebin' of https://github.com/02yebin/BE-springBasic into yebin

02yebin committed:
[BE] (DOCS) 위반 사례 업데이트

위반 사례를 발견하여 업데이트합니다.

02yebin committed:
[BE] (FEAT) UserDAO 인터페이스 생성

DIP 원칙을 지키기 위해 인터페이스를 만들었습니다.

02yebin committed:
Merge branch 'yebin' of https://github.com/02yebin/BE-springBasic into yebin

02yebin committed:
[BE] (FEAT) 할인정책을 구분할 enum 생성

할인 정책을 구분하기 위해 enum을 생성하였습니다.

02yebin committed:
[BE] (FEAT) DiscountPolicy 클래스에 메서드 추가

고정 금액 할인을 위한 메서드 생성

02yebin committed:
[BE] (CHORE) DiscountPolicy 클래스내 메서드 이름이 모호해 수정

정률 할인을 담당하는 메서드의 이름을 구분하기 쉽도록 바꾸어주었습니다.

02yebin committed:
[BE] (REFACTOR) DiscountPolicy 클래스내 할인된 가격을 계산하는 메서드 수정

할인 정책이 무엇인지 받아와 어떤 정책을 적용하는지 골라내어 상품 가격을 계산하고 반환합니다.

02yebin committed:
[BE] (REFACTOR) Order 클래스 수정

ItemOrder() 메서드를 호출할때마다 객체를 생성해야해서 바깥에 빼주었고, 할인 정책 구분을 위한 인자도 추가해주었습니다.

02yebin committed:
[BE] (perf) 1 WEEK 로 교체

[BE] (perf) 1 WEEK 로 교체

1주차 정답 코드로 교체합니다.
1주차 정답 코드를 수정해 커밋하겠습니다.

02yebin committed:
[ BE ] (feat) RateDiscountPolicy 클래스 생성

[ BE ] (feat) RateDiscountPolicy 클래스 생성

정률 할인 정책을 위한 클래스를 생성했습니다.
할인율을 변수에 담고 if 문으로 걸러내어 VIP이면 10%를 할인할 수 있도록 만들었습니다.

02yebin committed:
[ BE ] (feat) CalculatePrice 인터페이스 생성

[ BE ] (feat) CalculatePrice 인터페이스 생성

고정 금액 할인과 정률 할인 정책중 후에 바뀔수 있으므로 인터페이스를 만들어 클래스를 따로 빼내어 만들어주기 위해 상속받을 인터페이스를 생성했습니다.

02yebin committed:
[ BE ] (feat) CalculatePriceFix 클래스 생성

[ BE ] (feat) CalculatePriceFix 클래스 생성

고정 금액 할인 정책으로 바꿀 시 클래스만 바꾸어주도록 설계했습니다.

02yebin committed:
[ BE ] (feat) CalculatePriceRate 클래스 생성

[ BE ] (feat) CalculatePriceRate 클래스 생성

정률 할인 정책으로 바꿀 시 클래스만 바꾸어주도록 설계했습니다.

02yebin committed:
[ BE ] (pref) Order 클래스 수정

[ BE ] (pref) Order 클래스 수정

할인 정책의 변화에 따라 계산 법이 달라지므로 클래스만 바꾸어주어 계산하고 반환하도록 수정했습니다.

02yebin committed:
[ BE ] (docs) 위반 사례를 찾아서 해결책 제시 

[ BE ] (docs) 위반 사례를 찾아서 해결책 제시 

Order 클래스의 위반 사례 - OCP 위반을 찾아, 정률 할인와 고정 금액 할인 계산법을 따로 클래스로 빼내는 것으로 해결책을 제시했습니다.

02yebin committed:
[ BE ] (pref) Order 클래스의 변수 이름 수정

[ BE ] (pref) Order 클래스의 변수 이름 수정

변수이름과 메서드 이름이 겹쳐서 변수이름을 변경하였습니다.

02yebin committed:
Merge branch 'yebin' of https://github.com/02yebin/BE-springBasic into yebin

02yebin committed:
[ BE ] (pref) OrderService 클래스의 할인 정책 변경

[ BE ] (pref) OrderService 클래스의 할인 정책 변경

고정 금액 할인에서 정률 할인으로 바뀌었으므로 적용하는 메서드 또한, 바꾸어주었습니다.

02yebin committed:
[ BE ] (perf) RateDiscountPolicy 클래스 수정

[ BE ] (perf) RateDiscountPolicy 클래스 수정

할인율을 반환하는 것이 아닌, 할인된 금액을 반환하는 것이 맞다고 판단되어 수정하였습니다.

02yebin committed:
[ BE ] (perf) Order 클래스 수정

[ BE ] (perf) Order 클래스 수정

DiscountPolicy 클래스에서 할인한 가격을 반환하도록 수정을 하여 상품가격계산 메서드를 상품가격-할인한 가격 으로 계산하는 방식으로 수정하였습니다.

02yebin committed:
[ BE ] (perf) 할인율 관련 모든 자료형 수정

[ BE ] (perf) 할인율 관련 모든 자료형 수정

할인율 계산할때 소수점이들어가는데, int 자료형을 쓰니 0이 반환이 되어 float 형으로 바꾸어주었습니다.

02yebin committed:
[ BE ] (perf) Order 클래스 수정

[ BE ] (perf) Order 클래스 수정

calculatePrice() 메서드는 SRP 위반이라 생각되어 빼주었습니다.

02yebin committed:
[ BE ] (perf) OrderService 인터페이스 수정

[ BE ] (perf) OrderService 인터페이스 수정

Order 클래스에 있던 calculatePrice() 메서드를 빼와 서비스에 넣어주었습니다.

02yebin committed:
[ BE ] (perf) OrderServiceImpl 클래스 수정

[ BE ] (perf) OrderServiceImpl 클래스 수정

인터페이스에 calculatedPrice() 메서드를 추가해주어 구현해주었습니다.

02yebin committed:
[ BE ] (docs) Order 클래스의 위반 사항 수정

[ BE ] (docs) Order 클래스의 위반 사항 수정

Order 클래스가 OCP 위반이 아닌 SRP 위반으로 생각되어 수정하였습니다.

02yebin committed:
[ BE ] (docs) OCP 위반사례 발견, 해결책 추가

[ BE ] (docs) OCP 위반사례 발견, 해결책 추가

OCP 위반이라 생각되는 사례를 발견하여 해결책을 추가하였습니다.

02yebin committed:
[ BE ] (perf) 필요없는 클래스 삭제

[ BE ] (perf) 필요없는 클래스 삭제

CalculatePrcie 클래스가 필요없어서 삭제합니다.

02yebin committed:
Merge branch 'yebin' of https://github.com/02yebin/BE-springBasic into yebin

02yebin committed:
[ BE ] (perf) DependencyInjection 클래스 생성

[ BE ] (perf) DependencyInjection 클래스 생성

OCP 위반을 해결하기위해 의존성을 주입해주는 클래스를 따로 만들어주었습니다.

02yebin committed:
[ BE ] (perf) OrderServiceImpl 클래스 수정

[ BE ] (perf) OrderServiceImpl 클래스 수정

의존성을 주입해주는 클래스를 따로 만들어 그 클래스의 메서드를 사용하도록 수정하였습니다.

02yebin committed:
[ BE ] (perf) MemberServiceImpl 클래스 수정

[ BE ] (perf) MemberServiceImpl 클래스 수정

의존성 주입을 해주는 클래스를 따로 만들어 그 클래스의 메서드를 사용하도록 수정하였습니다.

02yebin committed:
[ BE ] (docs) 1 주차 HOME WORK 수정

[ BE ] (docs) 1 주차 HOME WORK 수정

1주차 homework답변 수정하였습니다.

02yebin committed:
[ BE ] (docs) 1주차 home work답변이 안보여서 수정

[ BE ] (docs) home work답변이 안보여서 수정

답변 쓴게 안보여서 수정하였습니다.

02yebin committed:
[ BE ] (perf) DependencyInjection 클래스 명 수정

[ BE ] (perf) DependencyInjection 클래스 명 수정

DependencyInjection 클래스명을 AppConfig로 수정하였습니다.

02yebin committed:
Merge branch 'yebin' of https://github.com/02yebin/BE-springBasic into yebin

02yebin committed:
[ BE ] (perf) Member 클래스 수정

[ BE ] (perf) Member 클래스 수정

어노테이션 Getter, Setter를 사용함에따라 getter, setter 관련 메서드들을 삭제하였습니다.

02yebin committed:
[ BE ] (perf) OrderApp 클래스 수정

[ BE ] (perf) OrderApp 클래스 수정

AppConfig 클래스로 의존주입 기능을 만들어 OrderApp 클래스에도 적용시켜 주었습니다.

02yebin committed:
[ BE ] (perf) MemeberApp 클래스 수정

[ BE ] (perf) MemeberApp 클래스 수정

AppConfig 클래스에서 의존 주입 기능을 만들어 MemberApp 클래스에도 적용시켜주었습니다.

02yebin committed:
[ BE ] (perf) OrderServiceImpl 클래스 수정

[ BE ] (perf) OrderServiceImpl 클래스 수정

AppConfig 클래스에서 의존주입 기능을 만들어 OrderServiceImpl 클래스에도 적용시켜주어 필요한 클래스를 생성자에서 초기화 하도록 만들었습니다.

02yebin committed:
[ BE ] (perf) MemberServiceImpl 클래스 수정

[ BE ] (perf) MemberServiceImpl 클래스 수정

AppConfig 클래스에서 의존주입 기능을 만들어 MemberServiceImpl 클래스에도 적용시켜주어 필요한 클래스를 생성자에서 초기화 하도록 만들었습니다.

02yebin committed:
[ BE ] (perf) OrderService 클래스 수정

[ BE ] (perf) OrderService 클래스 수정

빈등록해주고 연결해주었습니다.

02yebin committed:
[ BE ] (perf) RateDiscountPolicy 클래스 수정

[ BE ] (perf) RateDiscountPolicy 클래스 수정

빈 등록 해주었습니다.

02yebin committed:
[ BE ] (perf) MemberServiceImpl 클래스 수정

[ BE ] (perf) MemberServiceImpl 클래스 수정

빈 등록해주고 연결해주었습니다.

02yebin committed:
[ BE ] (perf) MemoryMemberRepository 클래스 수정

[ BE ] (perf) MemoryMemberRepository 클래스 수정

빈 등록해주었습니다.

02yebin committed:
[ BE ] (perf) 빈 자동 등록 어노테이션 수정

[ BE ] (perf) 빈 자동 등록 어노테이션 수정

빈 자동해주는 어노테이션을 모두 삭제하였습니다.

02yebin committed:
[ BE ] (perf) AppConfig 클래스 수정

[ BE ] (perf) AppConfig 클래스 수정

빈을 수동으로 등록하도록 어노테이션 Bean 을 사용하였습니다.

02yebin committed:
Merge branch 'FX-STUDY:main' into yebin

02yebin committed:
[ BE ] (feat) 새로운 프로젝트 생성

[ BE ] (feat) 새로운 프로젝트 생성

프로젝트를 새롭게 생성하여 4주차 미션은 이 프로젝트에 작업할 예정입니다.

02yebin committed:
[ BE ] (feat) Item 클래스 생성

[ BE ] (feat) Item 클래스 생성

상품의 필요한 정보를 담는 클래스를 생성하였습니다. 빌드패턴을 사용하여 생성자를 만들어주고, getter와 setter는 어노테이션으로 처리하였습니다.

02yebin committed:
[ BE ] (feat) ItemRepository 인터페이스 생성

[ BE ] (feat) ItemRepository 인터페이스 생성

상품을 저장할 save() 와 아이디로 상품을 찾을 findById() 를 만들 예정이므로 인터페이스에 생성해두었습니다.

02yebin committed:
[ BE ] (feat) ItemService 인터페이스 생성

[ BE ] (feat) ItemService 인터페이스 생성

상품을 저장하도록 시킬 join() 과 수정 시킬 update(), 상품을 찾아 반환할 findItem() 을 만들 예정이라 추가해두었습니다.

02yebin committed:
[ BE ] (perf) ItemPepository 어노테이션 추가

[ BE ] (perf) ItemPepository 어노테이션 추가

자동 등록 사용을 위한 어노테이션 Repository를 추가하였습니다.

02yebin committed:
[ BE ] (feat) MemoryItemRepository 클래스 생성

[ BE ] (feat) MemoryItemRepository 클래스 생성

ItemRepository를 상속받는 클래스를 생성하여 아이디와 상품을 저장할 저장소 store를 만들어주었고, 상품 번호  SEQUENCE를 만들어주었습니다.

02yebin committed:
[ BE ] (feat) MemoryItemRepository 클래스 수정

[ BE ] (feat) MemoryItemRepository 클래스 수정

save() 메서드를 오버라이딩하여 상품을 저장하는 메서드로 만들어주었습니다.

02yebin committed:
[ BE ] (feat) ItemRepository 클래스에 기능 추가

[ BE ] (feat) ItemRepository 클래스에 기능 추가

아이디로 상품을 찾아오는 findById를 오버라이딩하여 저장소 store에서 아이디로 상품을 찾아오도록 만들어주었습니다.

02yebin committed:
[ BE ] (perf) ItemService 어노테이션 추가

[ BE ] (perf) ItemService 어노테이션 추가

자동 등록을 위한 어노테이션 Service를 추가해주었습니다.

02yebin committed:
[ BE ] (feat) ItemServiceImpl 클래스 생성

[ BE ] (feat) ItemServiceImpl 클래스 생성

ItemRepository의 기능을 호출해야하므로 final 변수로 선언해주었고, 생성자에 Autowired 어노테이션을 사용하여 자동연결 해주었습니다.

02yebin committed:
[ BE ] (feat) ItemService 클래스에 join 메서드 구현

[ BE ] (feat) ItemService 클래스에 join 메서드 구현

상품 정보 저장을 지시하는 메서드를 구현하였습니다.

02yebin committed:
[ BE ] (feat) ItemService 클래스에 findItem 메서드 구현

[ BE ] (feat) ItemService 클래스에 findItem 메서드 구현

상품을 찾도록 지시하는 메서드를 구현하였습니다.

02yebin committed:
[ BE ] (perf) itemRepository 클래스 수정

[ BE ] (perf) itemRepository 클래스 수정

컨트롤러에서 save() 메서드를 호출하여 업데이트 방법을 고려해봤으나, 떠오르는 방법이 없어 update() 메서드를 추가해주었습니다.

02yebin committed:
[ BE ] (perf) MemoryItemRepository 클래스 수정

[ BE ] (perf) MemoryItemRepository 클래스 수정

상품 정보를 수정하는 메서드를 구현하였습니다.

02yebin committed:
[ BE ] (perf) ItemRepository 클래스 수정

[ BE ] (perf) ItemRepository 클래스 수정

상품 수정에 필요한 정보를 매개변수로 받아오는데 가시성이 떨어지는것 같아 Item 객체로 바꾸었습니다.

02yebin committed:
[ BE ] (perf) ItemService 클래스 수정

[ BE ] (perf) ItemService 클래스 수정

상품 수정에 필요한 매개변수를 서비스 쪽에서 받아오도록 했습니다.

02yebin committed:
[ BE ] (perf) MemoryItemRepository 클래스 수정

[ BE ] (perf) MemoryItemRepository 클래스 수정

저장소에 저장되어있는 상품을 id로 찾아 replace() 로 값을 바꾸어주었습니다.

02yebin committed:
[ BE ] (feat) itemServiceImpl 클래스에 updateItem 메서드 구현

[ BE ] (feat) itemServiceImpl 클래스에 updateItem 메서드 구현

상품 객체를 받아와 itemRepository id와 setter 로 수정한 상품 객체를 넘겼습니다.

서비스에서 setter로 정보를 바꾸는게 맞는가 싶습니다. 그렇다고 repository에서 많은 매개변수를 받아와 연산하는 것도 아닌것같고..

02yebin committed:
[ BE ] (perf) ItemRepository 클래스 어노테이션 수정

[ BE ] (perf) ItemRepository 클래스 어노테이션 수정

@Repository 가 인터페이스에 설정되어있어 구현체로 수정해주었습니다.

02yebin committed:
[ BE ] (perf) ItemService 클래스 어노테이션 수정

[ BE ] (perf) ItemService 클래스 어노테이션 수정

@Service 가 인터페이스에 설정되어있어 구현체로 수정해주었습니다.

02yebin committed:
[ BE ] (perf) MemoryItemRepository 클래스 어노테이션 추가

[ BE ] (perf) MemoryItemRepository 클래스 어노테이션 추가

인터페이스가 아닌 구현체에 @Repository 어노테이션을 추가하였습니다.

02yebin committed:
[ BE ] (feat) ItemListForm 작성

[ BE ] (feat) ItemListForm 작성

상품 리스트가 출력될 폼을 작성하였습니다.

02yebin committed:
[ BE ] (feat) ItemDetailForm 작성

[ BE ] (feat) ItemDetailForm 작성

상품 상세 폼을 작성하였습니다.

02yebin committed:
[ BE ] (feat) ItemSaveForm 작성

[ BE ] (feat) ItemSaveForm 작성

상품 등록 폼을 작성하였습니다.

02yebin committed:
[ BE ] (feat) ItemUpdateForm 작성

[ BE ] (feat) ItemUpdateForm 작성

상품 수정 폼을 작성하였습니다. 상품 ID 부분은 수정이 불가하기에 disabled 속성을 부여하였습니다.

02yebin committed:
[ BE ] (feat) ItemController 클래스 생성

[ BE ] (feat) ItemController 클래스 생성

itemController를 생성하여 url 매핑을 해주어 상품 목록폼을 화면에 띄우도록 하였습니다.

02yebin committed:
[ BE ] (feat) ItemController 클래스의 상품상세 매핑

[ BE ] (feat) ItemController 클래스의 상품상세 매핑

url 매핑을 해주어 상품 상세폼을 화면에 띄우도록 하였습니다.

02yebin committed:
[ BE ] (feat) ItemController 클래스의 상품등록 매핑

[ BE ] (feat) ItemController 클래스의 상품등록 매핑

url 매핑을 해주어 상품 등록폼을 화면에 띄우도록 하였습니다.

02yebin committed:
[ BE ] (feat) ItemController 클래스의 상품수정 매핑

[ BE ] (feat) ItemController 클래스의 상품수정 매핑

url 매핑을 하여 상품 수정폼을 화면에 띄우도록 하였습니다.

02yebin committed:
[ BE ] (perf) 상품 상세폼 버튼 수정

[ BE ] (perf) 상품 상세폼 버튼 수정

버튼을 누르면 해당 페이지로 이동하도록 수정하였습니다.

02yebin committed:
[ BE ] (perf) 상품 목록폼 버튼 수정

[ BE ] (perf) 상품 목록폼 버튼 수정

버튼을 누르면 해당 페이지로 이동하도록 수정하였습니다.

02yebin committed:
[ BE ] (perf) 상품 등록폼 버튼 수정

[ BE ] (perf) 상품 등록폼 버튼 수정

버튼을 누르면 해당 페이지로 이동하도록 수정하였습니다.

02yebin committed:
[ BE ] (perf) ItemController 수정

[ BE ] (perf) ItemController 수정

상품 상세폼에 ItemRepository가 필요해 선언해주었습니다.

02yebin committed:
[ BE ] (perf) ItemController 상품 상세폼 메서드 수정

[ BE ] (perf) ItemController 상품 상세폼 메서드 수정

상품 아이디를 받아와 정보를 띄워야해 @PathVariable로 id를 받아오도록 하였습니다.

02yebin committed:
[ BE ] (feat) Repository 클래스 findAll()메서드 추가

[ BE ] (feat) Repository 클래스 findAll()메서드 추가

상품 목록 폼에 상품들을 띄우려면 상품을 모두 찾아오는 메서드가 필요할 것 같아 추가하였습니다.

02yebin committed:
[ BE ] (feat) Repsitory 인터페이스에 findAll() 메서드 추가

[ BE ] (feat) Repsitory 인터페이스에 findAll() 메서드 추가

@Override 해주지않으니 findAll() 메서드를 찾지 못해 인터페이스에도 추가해주었습니다.

02yebin committed:
[ BE ] (perf) Repository 클래스 findAll() 메서드 오버라이딩

[ BE ] (perf) Repository 클래스 findAll() 메서드 오버라이딩

@Override 해주지않으니 findAll() 메서드를 찾지 못해 인터페이스에 메서드를 생성해주고 어노테이션을 추가해주었습니다.

02yebin committed:
[ BE ] (perf) ItemController 클래스 getItemListForm() 메서드 수정

[ BE ] (perf) ItemController 클래스 getItemListForm() 메서드 수정

상품 목록을 출력하려면 상품 정보를 받아와야하기에 findAll() 메서드를 생성하고 받아왔습니다.

02yebin committed:
[ BE ] (perf) Controller mapping수정

[ BE ] (perf) Controller mapping수정

버튼을 눌렀을 때 기능을 만들려고 보니 폼을 띄우는 메서드와 매핑 이름이 중복되어 폼을 띄우는 메서드는 모두 'Form"을 붙여주었습니다.

02yebin committed:
[ BE ] (perf) ItemSaveForm 수정

[ BE ] (perf) ItemSaveForm 수정

상품 등록 버튼에 href로 이동할 페이지 지정해놓았으나, 해당 버튼을 누르면 작동하는 메서드를 만들어 삭제했습니다.

02yebin committed:
[ BE ] (feat) Controller 클래스 itemSave 메서드 추가

[ BE ] (feat) Controller 클래스 itemSave 메서드 추가

상품 등록 버튼을 누르면 작동하는 메서드를 추가하였습니다. 상품을 저장하고 상품 목록을 띄우도록 만들었습니다.

02yebin committed:
[ BE ] (perf) Repository 인터페이스 수정

[ BE ] (perf) Repository 인터페이스 수정

상품 수정할때 아이디를 수정하지는 않으니 id를 굳이 받아올 필요가 없다고 생각해서 삭제했습니다.

02yebin committed:
[ BE ] (perf) Service 클래스 수정

[ BE ] (perf) Service 클래스 수정

상품을 업데이트할 때, 상품의 아이디가 수정되진않으니 id를 삭제하였습니다.

02yebin committed:
[ BE ] (perf) Repository 클래스 update 메서드 수정

[ BE ] (perf) Repository 클래스 update 메서드 수정

컨틀롤러에서 update() 메서드를 사용하려했으나, 기능이 Service에 구현되어 있어 update() 메서드에 기능을 구현하고, 수정하는 아이템의 id로 저장소에서 찾아와 정보를 바꾸기때문에 store.put() 을 할 필요가 없어보여 삭제했습니다.

02yebin committed:
[ BE ] (feat) Controller 클래스 itemUpdate() 메서드 추가

[ BE ] (feat) Controller 클래스 itemUpdate() 메서드 추가

상품 수정버튼을 눌렀을 때의 기능을 작성하였습니다.

02yebin committed:
[ BE ] (perf) ItemListForm 상품 등록 버튼 수정

[ BE ] (perf) ItemListForm 상품 등록 버튼 수정

상품 등록폼 매핑을 바꾸어 itemSaveForm 으로 수정해주었습니다.

02yebin committed:
[ BE ] (perf) ItemUpdateForm 취소 버튼 수정

[ BE ] (perf) ItemUpdateForm 취소 버튼 수정

취소 버튼을 눌렀을 때, 상품 목록으로 이동하도록 수정해주었습니다.

02yebin committed:
[ BE ] (perf) ItemController 클래스 getItemUpdateForm() 메서드 수정

[ BE ] (perf) ItemController 클래스 getItemUpdateForm() 메서드 수정

수정할 상품의 정보를 가지고 상품 수정폼으로 이동하도록 수정하였습니다.

02yebin committed:
[ BE ] (perf) Item 클래스 수정

[ BE ] (perf) Item 클래스 수정

생성자를 만들고싶었는데 void를 붙여놔서 생성자 기능을 못해 삭제해주었고, 상품 수량의 타입이 Long 일 필요까진 없을것 같아 int 로 수정하였습니다.

02yebin committed:
[ BE ] (perf) ItemService 인터페이스 updateItem() 메서드 매개변수 수정

[ BE ] (perf) ItemService 클래스 updateItem() 메서드 매개변수 수정

상품 수량의 타입을 int 로 수정해주었습니다.

02yebin committed:
[ BE ] (perf) ItemService 클래스 updateItem() 메서드 매개변수 수정

[ BE ] (perf) ItemService 클래스 updateItem() 메서드 매개변수 수정

상품 수량의 타입을 int 로 수정해주었습니다.

02yebin committed:
[ BE ] (perf) ItemListForm에 thymeleaf 추가

[ BE ] (perf) ItemListForm에 thymeleaf 추가

thymeleaf를 사용하여 상품 등록 버튼에 적용 시켜주었지만 페이지 이동을 하지 못합니다..

02yebin committed:
[ BE ] (perf) ItemListForm 상품 리스트 표시 부분 수정

[ BE ] (perf) ItemListForm 상품 리스트 표시 부분 수정

상품 리스트 표시 부분을 thymeleaf를 사용하여 나타내도록 수정하였습니다.

02yebin committed:
[ BE ] (perf) Controller 클래스 requestMapping 추가

[ BE ] (perf) Controller 클래스 requestMapping 추가

클래스 단위로 @RequestMapping 을 해주었습니다.

02yebin committed:
[ BE ] (perf) Controller 클래스 getItemDetailForm 매핑 수정

[ BE ] (perf) Controller 클래스 getItemDetailForm 매핑 수정

매핑한 주소가 세분화되면 좋을 것 같아 세분화 해주었습니다.

02yebin committed:
[ BE ] (perf) ItemDetailForm 수정

[ BE ] (perf) ItemDetailForm 수정

thymeleaf를 사용하여 상품의 상세 정보를 나타내도록 수정하였습니다.

02yebin committed:
[ BE ] (perf) Item 클래스 수정

[ BE ] (perf) Item 클래스 수정

int와 double 타입 대신 Integer 타입으로 null를 받아도 에러가 나지 않도록 수정하였습니다.

02yebin committed:
[ BE ] (perf) Repository 인터페이스 수정

[ BE ] (perf) Repository 인터페이스 수정

update() 메서드 기능을 수정하려고 보았더니 item 객체만 받아오면 새로 만든 item객체의 아이디를 사용하게 되어 기존의 상품 아이디와 달라 itemId 파라미터를 받아오도록 수정하였습니다.

02yebin committed:
[ BE ] (perf) Repository 인터페이스 수정

[ BE ] (perf) Repository 인터페이스 수정

테스트를 위해 Item 객체를 반환하도록 수정하였습니다.

02yebin committed:
[ BE ] (perf) Repository 인터페이스 수정

[ BE ] (perf) Repository 인터페이스 수정

테스트를 위해 저장소를 비우는 메서드를 추가하였습니다.

02yebin committed:
[ BE ] (perf) Repository 클래스 수정

[ BE ] (perf) Repository 클래스 수정

반환 타입을 수정하여 item 객체를 반환하도록 수정하였습니다.

02yebin committed:
[ BE ] (perf) Repository 클래스 수정

[ BE ] (perf) Repository 클래스 수정

기존에 수정한 item객체만 받아오도록 하였는데, 그렇게하면 기존의 상품 아이디를 알지 못하여 상품 아이디를 받아오도록 수정하였습니다.

02yebin committed:
[ BE ] (feat) Repository 클래스 수정

[ BE ] (feat) Repository 클래스 수정

저장소의 데이터를 모두 삭제하는 clearStore() 메서드를 추가하였습니다.

02yebin committed:
[ BE ] (perf) Controller 수정

[ BE ] (perf) Controller 수정

매핑 주소를 수정하고 url로 정보를 넘길 필요 없는 메서드는 post 매핑으로 수정해주었습니다.

02yebin committed:
[ BE ] (perf) Controller 수정

[ BE ] (perf) Controller 수정

@ModelAttribute 어노테이션을 이용하여 item 객체를 받아오도록 하고 상품을 저장하도록 수정

02yebin committed:
[ BE ] (perf) Controller 수정

[ BE ] (perf) Controller 수정

상품 아이디를 받아와 상품 객체를 찾도록 수정하였습니다.

02yebin committed:
[ BE ] (feat) Controller 수정

[ BE ] (feat) Controller 수정

@PostConstruct 어노테이션을 이용하여 초기에 상품을 세팅하여 확인할 수 있는 메서드를 추가해주었습니다.

02yebin committed:
[ BE ] (perf) ItemDetailForm.html 수정

[ BE ] (perf) ItemDetailForm.html 수정

th:text로 하였는데 input 위젯 밖에 텍스트가 써져 value로 수정하였습니다. 또한, item.get~~() 메서드를 사용하지 않고 item.id 이런식으로 정보를 받아올 수 있다는 것을 알게 되어 수정해주었습니다.

02yebin committed:
[ BE ] (perf) ItemDetailForm.html 수정

[ BE ] (perf) ItemDetailForm.html 수정

폼도 띄우고 컨트롤러에서 메서드를 어떻게 연결해주나 했더니 둘 다 써주면 되는 거였군요! type="button"을 써주지않으면 제 기능을 못해 추가해주었습니다.

02yebin committed:
[ BE ] (perf) ItemDetailForm.html 수정

[ BE ] (perf) ItemDetailForm.html 수정

타임리프를 이용하여 컨트롤러의 메서드를 연결시켜주고 type 설정을 해주었습니다.

02yebin committed:
[ BE ] (perf) ItemListform.html 수정

[ BE ] (perf) ItemListform.html 수정

상품 등록 버튼의 기존 매핑주소를 바꿔 href를 수정해주고 type을 지정해주었습니다.

02yebin committed:
[ BE ] (perf) ItemListform.html 수정

[ BE ] (perf) ItemListform.html 수정

a 태그의 href를 추가하여 상품 상세 폼을 가져오도록 수정하고 .get~~() 메서드 말고 .Id 등과 같이 수정해주었습니다.

02yebin committed:
[ BE ] (perf) ItemSaveForm.html 수정

[ BE ] (perf) ItemSaveForm.html 수정

상품 등록 버튼의 type을 submit으로 수정해주고, 취소 버튼을 누르면 상품 리스트 폼으로 이동할 수 있도록 타임리프를 이용해 수정하였습니다.

02yebin committed:
[ BE ] (perf) ItemUpdateForm.html 수정

[ BE ] (perf) ItemUpdateForm.html 수정

th:value 를 이용하여 상품의 정보들을 표시하도록 수정하였습니다.

02yebin committed:
[ BE ] (perf) ItemUpdateForm.html 수정

[ BE ] (perf) ItemUpdateForm.html 수정

저장 버튼의 type을 submit으로 지정해주었습니다.

02yebin committed:
[ BE ] (perf) ItemUpdateForm.html 수정

[ BE ] (perf) ItemUpdateForm.html 수정

타임리프를 이용해 취소 버튼을 눌렀을 때, 상품 리스트 폼으로 이동하도록 수정해주었습니다.

02yebin committed:
[ BE ] (feat) Repository Test

[ BE ] (feat) Repository Test

테스트를 수행 한 후, 저장소 초기화를 위해 @afterEach 어노테이션을 사용하여 자동 초기화 되도록 만들어주었습니다.

02yebin committed:
[ BE ] (feat) repository test

[ BE ] (feat) repository test

save 메서드를 테스트 해보았습니다. 정상 작동합니다.

02yebin committed:
[ BE ] (feat) repository test

[ BE ] (feat) repository test

findAll 메서드 테스트를 해보았습니다.

02yebin committed:
[ BE ] (feat) repository test

[ BE ] (feat) repository test

updaet 메서드 테스트를 해보았습니다.

02yebin committed:
[ BE ] (perf) ItemSaveForm.html 수정

[ BE ] (perf) ItemSaveForm.html 수정

상품 정보를 입력하고 등록을 하기위해 form에 action 을 설정해주었고, name 속성이 Item 클래스가 가진 변수의 이름과 동일해야 @ModelAttribute 로 가져올 수 있어 수정해주었습니다.

02yebin committed:
[ BE ] (feat) index.html 생성

[ BE ] (feat) index.html 생성

redirect:/ 를 사용하려고하는데 자꾸 에러가나서 찾아보니 index 파일이 필요하다고 하여 생성하였습니다.

02yebin committed:
[ BE ] (perf) ItemDetailForm.html 수정

[ BE ] (perf) ItemDetailForm.html 수정

name 속성을 Item 객체의 변수의 이름과 동일하게 수정해주었습니다.

02yebin committed:
[ BE ] (chore) Controller 수정

[ BE ] (chore) Controller 수정

Item 으로 정보를 보내주는게 불편해 item으로 수정해주었습니다.

02yebin committed:
[ BE ] (perf) Controller 수정

[ BE ] (perf) Controller 수정

@PathVariable 로 원래 상품의 아이디를 가져오고 @ModelAttribute로 상품 객체를 알아서 만들어가져오도록 지정해주고, redirect:/ 이용하여 요청할 주소를 응답결과로 전달하도록 만들어주었습니다.

02yebin committed:
[ BE ] (perf) ItemUpdateForm.html 수정

[ BE ] (perf) ItemUpdateForm.html 수정

addAttribute를 item으로 수정해주어 Item->item 으로 수정해주었고, name 속성은 Item 객체의 변수 이름과 동일하게 맞추어 수정해 @ModelAttribute를 사용할 수 있게끔 만들어주었습니다.

02yebin committed:
Merge branch 'FX-STUDY:main' into yebin

02yebin committed:
[BE ] (docs)스프링 핵심 원리 기본편 8-9강 요약

스프링 핵심 원리 기본편 8-9강 요약

8-9강의 핵심내용 빈에 대하여 요약/정리 하였습니다.

02yebin committed:
Merge branch 'FX-STUDY:main' into yebin

02yebin committed:
[ BE ] (docs) MVC 1-4강 요약

[ BE ] (docs) MVC 1-4강 요약

MVC 1-4강에 대하여 중요하게 생각한 부분만 요약하여 정리하였습니다.

02yebin committed:
Merge branch 'FX-STUDY:main' into yebin

02yebin committed:
[ BE ] (docs) 스프링 MVC 5-7장 요약

[ BE ] (docs) 스프링 MVC 5-7장 요약

중요하고 알아둬야겠다고 생각되는 부분만 요약 정리하였습니다.

shinywoon commented:
좋습니다.
앞으로 지속적으로 사용해야 할 기술들 입니다.
확실한 숙지가 필요합니다.

02yebin committed:
Merge branch 'FX-STUDY:main' into yebin

02yebin committed:
[ BE ] (feat) 초기 세팅

[ BE ] (feat) 초기 세팅

초기 세팅입니다.

02yebin committed:
[ BE ] (perf) students 객체 생성

[ BE ] (perf) students 객체 생성

@Data 어노테이션을 사용하는 students 객체를 생성하였습니다.

02yebin committed:
[ BE ] (perf) Students 객체의 id 타입 변경

[ BE ] (perf) Students 객체의 id 타입 변경

제 자신의 학번을 생각하며 Integer로 만들었으나, 1 부터 시작하는 것을 고려해
Integer -> Long 으로 수정하였습니다.

02yebin committed:
[ BE ] (feat) repository 생성

[ BE ] (feat) repository 생성

repository를 생성하여 저장소와 아이디가 될 SEQUENCE를 만들었습니다.

02yebin committed:
[ BE ] (feat) 학생 정보 저장 메서드 생성

[ BE ] (feat) 학생 정보 저장 메서드 생성

학생 정보를 저장하는 메서드 save()를 만들어 학번은 1부터 늘어나도록 해주었습니다.

02yebin committed:
[ BE ] (feat) 학번으로 학생 찾는 메서드 생성

[ BE ] (feat) 학번으로 학생 찾는 메서드 생성

학생 한명을 학번으로 찾을 수 있도록 findById() 메서드를 생성하였습니다.

02yebin committed:
[ BE ] (feat) 학생 전부를 찾을 수 있는 메서드 생성

[ BE ] (feat) 학생 전부를 찾을 수 있는 메서드 생성

학생 전부를 찾을 수 있는 findAll() 메서드를 생성하였습니다.

02yebin committed:
[ BE ] (feat) 학생 정보 수정 메서드 생성

[ BE ] (feat) 학생 정보 수정 메서드 생성

학생 정보를 수정하는 update() 메서드를 생성하여 이름, 학년, 전공을 수정 가능하도록 만들었습니다.

02yebin committed:
[ BE ] (feat) 학생 정보 삭제 메서드 생성

[ BE ] (feat) 학생 정보 삭제 메서드 생성

학생 한명의 정보를 삭제하는 delete() 메서드를 생성하여 id를 이용하여 학생 정보를 찾아 삭제하도록 만들었습니다.

02yebin committed:
[ BE ] (feat) 학생 정보 모두 삭제 메서드 생성

[ BE ] (feat) 학생 정보 모두 삭제 메서드 생성

학생 정보를 모두 삭제하는 메서드 clearStore()를 생성하였습니다.

02yebin committed:
[ BE ] (test) repository test

[ BE ] (test) repository test

StudentRepository 테스트를 작성하였습니다. 테스트에 문제가 발생하지 않도록 @AfterEach를 이용하여 저장소를 비워주었습니다.

02yebin committed:
[ BE ] (test) save() test

[ BE ] (test) save() test

save() 메서드를 테스트하여 정상 작동하였습니다.

02yebin committed:
[ BE ] (test) findAll() test

[ BE ] (test) findAll() test

findAll() 메서드를 테스트하여 정상 작동하였습니다.

02yebin committed:
[ BE ] (test) update() test

[ BE ] (test) update() test

update() 메서드를 테스트하여 정상작동하였습니다.

02yebin committed:
[ BE ] (test) delete() test

[ BE ] (test) delete() test

delete() 메서드를 테스트하여 정상작동 하였습니다.

02yebin committed:
[ BE ] (perf) id 파라미터 삭제

[ BE ] (perf) id 파라미터 삭제

학번을 자동으로 생성하여 넣어주기때문에 id 파라미터를 삭제하였습니다.

02yebin committed:
[ BE ] (feat) service interface 생성

[ BE ] (feat) service interface 생성

학생 정보를 생성하여 저장한 join(),
학생 정보를 찾을 findStudent(),
학생 정보를 삭제할 deleteStudent(),
학생 정보를 수정할 updateStudent(),
다중 학생 정보를 삭제할 multiDeleteStudent() 메서드를 만들 예정입니다.

02yebin committed:
[ BE ] (feat) service impl 생성

[ BE ] (feat) service impl 생성

@Service, @RequiredArgsConstructor 어노테이션을 이용하여 repository를 등록시켜주었습니다.

02yebin committed:
[ BE ] (feat) join() 메서드 생성

[ BE ] (feat) join() 메서드 생성

repository의 save() 메서드를 이용하여 학생 정보를 저장하도록 만들었습니다.

02yebin committed:
[ BE ] (feat) findStudent() 메서드 생성

[ BE ] (feat) findStudent() 메서드 생성

학번으로 학생 정보를 찾는 메서드를 만들었습니다.

02yebin committed:
[ BE ] (feat) deleteSutdent() 메서드 생성

[ BE ] (feat) deleteSutdent() 메서드 생성

학번으로 단일 학생 정보를 삭제하는 메서드를 만들었습니다.

02yebin committed:
[ BE ] (feat) updateStudent() 메서드 생성

[ BE ] (feat) updateStudent() 메서드 생성

학번으로 학생을 찾아 정보를 수정하는 메서드를 만들었습니다.

02yebin committed:
[ BE ] (feat) multiDeleteStudent() 메서드 생성

[ BE ] (feat) multiDeleteStudent() 메서드 생성

foreach를 사용하여 입력된 학번의 학생 정보를 모두 삭제하는 메서드를 만들었습니다.

shinywoon reviewed (commented):
실력이 많이 올라온것 같아 뿌듯하네요
하지만 아쉬운건 아직 협업을 고려하지 않은 단계라는 느낌이 강합니다.
또한, logic 에서 발생 될 수 있는 예외 상황을 고려하지 않은 모습이 보입니다.
예를 들어 회원 가입 하려는 학생 정보가 이미 존재한다면? 
deleteStudent 이후 정상 삭제 여부 ? , 삭제 실패 시 어떤 정보를 Client 에 제공해 줄까? 등 여러 문제를 고민해야 합니다.

02yebin committed:
[ BE ] (perf) Students 클래스 수정

[ BE ] (perf) Students 클래스 수정

빌더패턴을 사용하여 @Data -> @Getter, @Setter로 수정하였습니다.
또한, 학번은 사용자에게 입력받도록 생성자에 추가하였습니다.

02yebin committed:
[ BE ] (perf) repository 수정

[ BE ] (perf) repository 수정

사용자에게 학번을 입력받도록 하여 SEQUENCE 를 삭제하였습니다.

02yebin committed:
[ BE ] (perf) update() 메서드 수정

[ BE ] (perf) update() 메서드 수정

코멘트보고 수정하였는데 제가 잘 이해한게 맞나요??

02yebin committed:
[ BE ] (perf) service 메서드명 수정

[ BE ] (perf) service 메서드명 수정

다른 분들이 메서드 명을 보고 어떤 기능의 메서드인지 알기 쉽도록 메서드명을 모두 수정하였습니다.

02yebin committed:
[ BE ] (perf) multiDeleteStudentInForById() 메서드 수정

[ BE ] (perf) multiDeleteStudentInForById() 메서드 수정

학번으로 학생의 정보가 잘 삭제되었으면 true를 안되었으면 false를 반환하도록 수정하였습니다.

shinywoon commented on 10 WEEK/students/src/main/java/yebin/students/StudentService.java:
```diff
@@ -4,8 +4,8 @@
-    public void deleteStudent(Long studentId);
-    public void updateStudent(Long studentId, Students updateStudent);
-    public void multiDeleteStudent(List<Long> studentId);
+    public Students showStudentInForById(Long studentId);
```
제 개인적인 취향으로는 
```java
public Studnets showStudentInformationBy(Long studentId);
```
이런식으로 작명합니다.

shinywoon reviewed (commented):
항상 협업을 염두하고 누군가 제 코드를 쉽게 읽고 해석할 수 있도록 신경써서 코딩해야합니다.
앞으로 더 좋은 방향의 발전을 기원합니다.

02yebin committed:
Merge branch 'FX-STUDY:main' into yebin

02yebin committed:
[ BE ] (build) 12 WEEK

[ BE ] (build) 12 WEEK

02yebin committed:
[ BE ] (feat) 학생 정보에 관한 enum 클래스 생성

[ BE ] (feat) 학생 정보에 관한 enum 클래스 생성

학생 정보에 필요하면서 제한적인 정보들을 enum 클래스로 생성해주었습니다.

02yebin committed:
[ BE ] (feat) 학생 클래스 생성

[ BE ] (feat) 학생 클래스 생성

학생의 정보를 담을 클래스를 생성하였습니다.
회원가입을 할땐 signUpFormData() 메서드를 회원정보를 수정할땐 updateFormData() 메서드를 호출하도록 하기위해 메서드를 따로 생성했습니다.

02yebin committed:
[ BE ] (feat) repository 생성

[ BE ] (feat) repository 생성

저장소로 Map을 사용하였습니다.

02yebin committed:
[ BE ] (feat) repository save() 기능 구현

[ BE ] (feat) repository save() 기능 구현

학생정보를 저장하는 메서드를 구현하였습니다.

02yebin committed:
[ BE ] (feat) repository findById() 기능 구현

[ BE ] (feat) repository findById() 기능 구현

학생의 정보를 학번으로 찾도록하는 메서드를 구현하였습니다.

02yebin committed:
[ BE ] (feat) repository findAll() 기능 구현

[ BE ] (feat) repository findAll() 기능 구현

메모리에 저장되어있는 모든 학생의 정보를 가져오는 메서드를 구현하였습니다.

02yebin committed:
[ BE ] (feat) repository updateStudentInfomation() 기능 구현

[ BE ] (feat) repository updateStudentInfomation() 기능 구현

학생의 정보를 수정하는 메서드에 활용할 기능 메서드를 구현하였습니다.

02yebin committed:
[ BE ] (feat) repository update() 기능 구현

[ BE ] (feat) repository update() 기능 구현

학생의 정보를 수정하는 메서드를 구현하였습니다.

02yebin committed:
[ BE ] (feat) service 클래스 생성

[ BE ] (feat) service 클래스 생성

제약조건을 걸어 각각 다른 정보를 반환하기 위해 Object 자료형으로 반환형식을 설정해두었습니다.

02yebin committed:
[ BE ] (feat) service 구현 클래스 생성

[ BE ] (feat) service 구현 클래스 생성

02yebin committed:
[ BE] (feat) service saveStudent() 구현

[ BE] (feat) service saveStudent() 구현

회원가입을 진행할 때 어떠한 부분이 요구사항을 충족시키지 못했는지 알려주기위해 해당 정보를 반환하도록 만들었습니다.

02yebin committed:
[ BE ] (feat) service showStudentByStudentId() 구현

[ BE ] (feat) service showStudentByStudentId() 구현

찾으려는 학생의 정보가 없으면 null을 반환하도록 만들었습니다.

02yebin committed:
[ BE ] (feat) service showAllStudents() 구현

[ BE ] (feat) service showAllStudents() 구현

저장소에 학생 정보가 아무것도 없으면 null 을 반환하도록 만들었습니다.

02yebin committed:
[ BE ] (feat) service updateStudentByStudentId() 구현

[ BE ] (feat) service updateStudentByStudentId() 구현

수정하려는 학생의 정보가 없다면 null을 반환하고 수정하려는 정보가 요구사항을 충족시키지 못했다면 해당 정보를 반환해주도록 만들었습니다.

02yebin committed:
[ BE ] (feat) 제약 조건 클래스 생성

[ BE ] (feat) 제약 조건 클래스 생성

학번, 이름, 학과, 학기, 학년에 관한 제약 조건을 확인하는 메서드를 만들 예정입니다.

02yebin committed:
[ BE ] (feat) 학번 제약 조건 확인 메서드 생성

[ BE ] (feat) 학번 제약 조건 확인 메서드 생성

학번의 길이가 7인지, 숫자로 이루어져있는지 확인하는 메서드를 생성하였습니다.

02yebin committed:
[ BE ] (feat) 이름 제약 조건 확인 메서드 생성

[ BE ] (feat) 이름 제약 조건 확인 메서드 생성

이름이 한글로 이루어져있는지, 2글자 이상 7글자 이하인지 확인하는 메서드를 생성하였습니다.

02yebin committed:
[ BE ] (feat) 학과 제약 조건 확인 메서드 생성

[ BE ] (feat) 학과 제약 조건 확인 메서드 생성

학과가 소프트웨어, 컴퓨터공학, 산업디자인학과 중 하나인지 확인하는 메서드를 생성하였습니다.

02yebin committed:
[ BE ] (feat) 학기 제약 조건 확인 메서드 생성

[ BE ] (feat) 학기 제약 조건 확인 메서드 생성

학기가 1학기, 2학기 중 하나인지 확인하는 메서드를 생성하였습니다.

02yebin committed:
[ BE ] (feat) 학년 제약 조건 확인 메서드 생성

[ BE ] (feat) 학년 제약 조건 확인 메서드 생성

학년이 1~5학년인지 확인하는 메서드를 생성하였습니다.

02yebin committed:
[ BE ] (feat) controller 클래스 생성

[ BE ] (feat) controller 클래스 생성

매핑은 /student 로 해주었습니다.

02yebin committed:
[ BE ] (feat) controler addStudent() 구현

[ BE ] (feat) controler addStudent() 구현

학생정보를 저장하는 메서드를 호출하는 메서드를 생성하였습니다.

02yebin committed:
[ BE ] (feat) controler showStudentByStudentId() 구현

[ BE ] (feat) controler showStudentByStudentId() 구현

학번으로 학생을 찾는 메서드를 호출하고 그 정보를 넘겨주는 메서드를 생성하였습니다.

02yebin committed:
[ BE ] (feat) controler showAllStudents() 구현

[ BE ] (feat) controler showAllStudents() 구현

모든 학생의 정보를 찾는 메서드를 호출하고 그 정보를 넘겨주는 메서드를 생성하였습니다.

02yebin committed:
[ BE ] (feat) controler updateStudentByStudentId() 구현

[ BE ] (feat) controler updateStudentByStudentId() 구현

학생의 정보를 수정하는 메서드를 호출하는 메서드를 생성하였습니다.
